tests/secrets/cat.test.ts

```diff
@@ -0,0 +1,148 @@
+import * as testUtils from '../utils';
+
+describe('commandCatSecret', () => {
+  const password = 'password';
+  const logger = new Logger('CLI Test', LogLevel.WARN, [new StreamHandler()]);
+  let dataDir: string;
+  let polykeyAgent: PolykeyAgent;
+
+  beforeEach(async () => {
+    dataDir = await fs.promises.mkdtemp(
+      path.join(globalThis.tmpDir, 'polykey-test-'),
+    );
+    polykeyAgent = await PolykeyAgent.createPolykeyAgent({
+      password,
+      options: {
+        nodePath: dataDir,
+        agentServiceHost: '127.0.0.1',
+        clientServiceHost: '127.0.0.1',
+        keys: {
+          passwordOpsLimit: keysUtils.passwordOpsLimits.min,
+          passwordMemLimit: keysUtils.passwordMemLimits.min,
+          strictMemoryLock: false,
+        },
+      },
+      logger: logger,
+    });
+  });
+  afterEach(async () => {
+    await polykeyAgent.stop();
+    await fs.promises.rm(dataDir, {
+      force: true,
+      recursive: true,
+    });
+  });
+
+  test('should retrieve a secret', async () => {
+    const vaultName = 'Vault3' as VaultName;
+    const vaultId = await polykeyAgent.vaultManager.createVault(vaultName);
+
+    await polykeyAgent.vaultManager.withVaults([vaultId], async (vault) => {
+      await vaultOps.addSecret(vault, 'MySecret', 'this is the secret');
+    });
+
+    const command = ['secrets', 'cat', '-np', dataDir, `${vaultName}:MySecret`];
+
+    const result = await testUtils.pkStdio(command, {
+      env: { PK_PASSWORD: password },
+      cwd: dataDir,
+    });
+    expect(result.exitCode).toBe(0);
+    expect(result.stdout).toBe('this is the secret');
+  });
+
+  test('should concatenate multiple secrets', async () => {
+    const vaultName = 'Vault3' as VaultName;
+    const vaultId = await polykeyAgent.vaultManager.createVault(vaultName);
+    const secretNames = ['secret-name1', 'secret-name2'];
+
+    await polykeyAgent.vaultManager.withVaults([vaultId], async (vault) => {
+      for (const secretName of secretNames) {
+        await vaultOps.addSecret(vault, secretName, secretName);
+      }
+    });
+
+    const secretPaths = secretNames.map((v) => `${vaultName}:${v}`);
+    const command = ['secrets', 'cat', '-np', dataDir, ...secretPaths];
+
+    const result = await testUtils.pkStdio(command, {
+      env: { PK_PASSWORD: password },
+      cwd: dataDir,
+    });
+    expect(result.exitCode).toBe(0);
+    expect(result.stdout).toBe(secretNames.join(''));
+  });
+
+  test('should concatenate secrets from multiple vaults', async () => {
+    const vaultNames = ['Vault3-1', 'Vault3-2'];
+    const vaultIds: Array<VaultId> = [];
+    for (const vaultName of vaultNames) {
+      vaultIds.push(await polykeyAgent.vaultManager.createVault(vaultName));
+    }
+    const secretNames = ['secret-name1', 'secret-name2', 'secret-name3'];
+    const secretVaultNames = [
+      [0, secretNames[0]],
+      [1, secretNames[1]],
+      [0, secretNames[2]],
+    ];
```

Comment:
This is a very confusing way to structure all your test data. It's better to keep it all flattened and simple. You don't need an array and loop over it to create vaults. You can just keep each vault name in its own variable, create each vault manually and save their Ids in their own variables. `vault1`, `vaultId1` etc etc/